server/src/main/kotlin/Server.kt

```diff
@@ -39,15 +38,13 @@ class Server {
 			return
 		}
 
-		if (reader.readByte() != PacketId.NAMECHANGE.value) {
-			error("first packet must be a namechange")
-		}
-		val nameChange = reader.readPacketNameChange()
+		val login = handleLogin(reader, writer) ?: return client.dispose()
```

Comment:
inline handleLogin as this is the only occurrence

Reply:
inlining would reduce testatibilty, abstraction, and understandability of the code.